TheMovieDb/Views/HomeViewController/HomeViewController.swift

```diff
@@ -0,0 +1,195 @@
+//
+//  HomeCollectionViewController.swift
+//  TheMovieDb
+//
+//  Created by Jonathan Hernandez on 10/11/21.
+//
+
+import UIKit
+
+class HomeViewController: UIViewController, HomeView {
+    
+    var arrMovies: [SectionMovie: [Movie]]!
+    var dataSource: UICollectionViewDiffableDataSource<SectionMovie, Movie>! = nil
+    let compositionalLayout: UICollectionViewCompositionalLayout = {
+        let itemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1),
+                                              heightDimension: .fractionalHeight(1))
+        let item = NSCollectionLayoutItem(layoutSize: itemSize)
+        
+        let groupFractionalWidth = 0.33
+        let groupFractionalHeight: Float = 0.30
+        let groupSize = NSCollectionLayoutSize(
+            widthDimension: .fractionalWidth(CGFloat(groupFractionalWidth)),
+            heightDimension: .fractionalHeight(CGFloat(groupFractionalHeight)))
+        let group = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, subitem: item, count: 1)
+        group.contentInsets = NSDirectionalEdgeInsets(top: 5, leading: 5, bottom: 5, trailing: 5)
+        
+        let headerSize = NSCollectionLayoutSize(
+          widthDimension: .fractionalWidth(1.0),
+          heightDimension: .estimated(44))
+        let sectionHeader = NSCollectionLayoutBoundarySupplementaryItem(
+          layoutSize: headerSize,
+          elementKind: "HeaderCollectionView.reuseIdentifier",
+          alignment: .top)
+        
+        let section = NSCollectionLayoutSection(group: group)
+        section.boundarySupplementaryItems = [sectionHeader]
+        section.orthogonalScrollingBehavior = .groupPaging
+        
+        return UICollectionViewCompositionalLayout(section: section)
+    }()
+    
+    private var latestSearch: String?
+    lazy private var searchController: SearchBar = {
+        let searchController = SearchBar("Search a Movie", delegate: self)
+        searchController.text = latestSearch
+        searchController.showsCancelButton = !searchController.isSearchBarEmpty
+        return searchController
+    }()
+    
+    var movieHomeCollectionView: UICollectionView!
```

Comment:
Same here: Please avoid force unwrap properties "!"

You can declare this view like this: `private lazy var movieHomeCollectionView: UICollectionView = { }()`
The same you did with the `searchController`.